tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_horizontal_execution_merging.py

```diff
@@ -172,3 +175,43 @@ def test_on_the_fly_merging_repeated():
     transformed = OnTheFlyMerging().visit(testee)
     hexecs = transformed.vertical_loops[0].sections[0].horizontal_executions
     assert len(hexecs) == 2
+
+
+def test_on_the_fly_merging_localscalars():
```

Comment:
@havogt the test is still here. I think I had an older version of this test too, but it wasn't really testing what it claimed to.